### ISSUE TITLE
Probable bug with Shanna, Purifying Blade

### ISSUE THREAD

**Issue by TheLastNarwhal** [Card-Forge/forge#3374]
**Describe the bug**
The A.I. played a land and cast Shanna, it did nothing else that could have gained them life. At eot they paid four for Shanna's trigger and I assume they drew, because my Orcish Bowmasters triggered four times. ~~I'm 90% sure no life gain occurred, but since I wasn't watching like a hawk I can't speak in absolutes...~~ 

On a following turn I carefully observed no life gain and then the A.I. paid two for Shanna's ability and drew two cards.

**Expected behavior**
If the A.I. paid for Shanna's ability when no life gain occurred, then nothing should happen.

**Screenshots**
Here are the screen shots of the field and log.

![Screenshot_20230630_185330](https://github.com/Card-Forge/forge/assets/92389638/e898d578-69df-4515-9bc0-ea0d95d88218)

![Screenshot_20230630_185408](https://github.com/Card-Forge/forge/assets/92389638/9ed105c9-dd9a-40f2-94a6-cc7fd975d69b)

**Comment by dracontes** [comment]
Checked this one on `desktop-1.6.64-SNAPSHOT-08-05` and Shanna, Purifying Blade works as expected both on my side and on the AI's side. I went for several turn cycles beyond what's displayed on the screenshot without anything relating to Shanna being amiss. To be on the safe side, I tested if the AI would react properly to gaining life by casting [Invigorate](https://scryfall.com/card/2xm/172/invigorate) via alternate cost, and it did so.

![shanna](https://github.com/user-attachments/assets/77ccf6ea-6edf-4a57-b5d3-69de0ddb9c0f)
_No drawing cards without life gain._

![shanna_invigorate](https://github.com/user-attachments/assets/4db68e75-10c2-4b92-abcc-3dc912e16bb1)
_AI drawing a card with life gain._
